apps/rabbitmq.go

```diff
@@ -16,9 +16,92 @@ package apps
 
 import (
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator"
+	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/fluentbit"
 	"github.com/GoogleCloudPlatform/ops-agent/confgenerator/otel"
 )
 
+type LoggingProcessorRabbitmq struct {
+	confgenerator.ConfigComponent `yaml:",inline"`
+}
+
+func (*LoggingProcessorRabbitmq) Type() string {
+	return "rabbitmq"
+}
+
+func (p *LoggingProcessorRabbitmq) Components(tag, uid string) []fluentbit.Component {
+	c := []fluentbit.Component{}
+	regexParser := confgenerator.LoggingProcessorParseRegex{
+		// Sample log line:
+		// 2022-01-31 18:01:20.441571+00:00 [erro] <0.692.0> ** Connection attempt from node 'rabbit_ctl_17@keith-testing-rabbitmq' rejected. Invalid challenge reply. **
+		Regex: `^(?<timestamp>\d+-\d+-\d+ \d+:\d+:\d+\.\d+\+\d+:\d+) \[(?<severity>\w+)\] \<(?<process_id>\d+\.\d+\.\d+)\> (?<message>.*)$`,
+		ParserShared: confgenerator.ParserShared{
+			TimeKey:    "timestamp",
+			TimeFormat: "%Y-%m-%d %H:%M:%S.%L+%Z",
+		},
+	}
+	c = append(c, regexParser.Components(tag, uid)...)
+
+	// severities documented here: https://www.rabbitmq.com/logging.html#log-levels
+	c = append(c, fluentbit.TranslationComponents(tag, "severity", "logging.googleapis.com/severity", true, []struct {
+		SrcVal  string
+		DestVal string
+	}{
+		{
+			"debug", "DEBUG",
+		},
+		{
+			"error", "ERROR",
+		},
+		{
+			"info", "INFO",
+		},
+		{
+			"noti", "DEFAULT",
+		},
+	})...)
+
+	return c
+}
+
+type LoggingReceiverRabbitmq struct {
+	LoggingProcessorRabbitmq                `yaml:",inline"`
+	confgenerator.LoggingReceiverFilesMixin `yaml:",inline" validate:"structonly"`
+}
+
+func (r LoggingReceiverRabbitmq) Components(tag string) []fluentbit.Component {
+	if len(r.IncludePaths) == 0 {
+		r.IncludePaths = []string{
+			"/var/log/rabbitmq/rabbit*.log",
+		}
+	}
+	// Some multiline entries related to crash logs are important to capture and end in
+	//
+	// 2022-01-31 18:07:43.557042+00:00 [erro] <0.130.0>
+	// BOOT FAILED
+	// ===========
+	// ERROR: could not bind to distribution port 25672, it is in use by another node: rabbit@keith-testing-rabbitmq
+	//
+	r.MultilineRules = []confgenerator.MultilineRule{
+		{
+			StateName: "start_state",
+			NextState: "cont",
+			Regex:     `^(?<timestamp>\d+-\d+-\d+ \d+:\d+:\d+\.\d+\+\d+:\d+) \[(?<severity>\w+)\] \<(?<process_id>\d+\.\d+\.\d+)\> (?<message>.*)$`,
+		},
+		{
+			StateName: "cont",
+			NextState: "cont",
+			Regex:     `^\s*$`,
```

Comment:
In the example in the comments above, will all of 
```
2022-01-31 18:07:43.557042+00:00 [erro] <0.130.0>
BOOT FAILED
===========
ERROR: could not bind to distribution port 25672, it is in use by another node: rabbit@keith-testing-rabbitmq

```

Be a single LogEntry? The regex multiline rules here suggest that they won't be no?

If we wanted them to be the same LogEntry would this make more sense?:
```
r.MultilineRules = []confgenerator.MultilineRule{
		{
			StateName: "start_state",
			NextState: "cont",
			Regex:     `\d+-\d+-\d+ \d+:\d+:\d+\.\d+\+\d+:\d+`,
		},
		{
			StateName: "cont",
			NextState: "cont",
			Regex:     `^(?!\d+-\d+-\d+ \d+:\d+:\d+\.\d+\+\d+:\d+)`,
		},
	}
```

Reply:
Ahh that makes more sense, I think I may have misunderstood quite how multiline was working but just tested this out and it appears to work great! Thanks for the suggestion